### PR TITLE
Implement OpenGraph TagHelpers.

### PR DESCRIPTION
**Pull request by Choc13**
Addresses #17 

Summary:
* Remove `HeaderMetadataViewComponent`.
* Replace Open Graph tags from `HeaderMetadataViewComponent` with a more comprehensive set of tag helpers for the major Open Graph object types.
* Deprecate robots head meta tags.

### CODE REVIEW

**Comment by Choc13** [comment]
Appreciate this is a large PR. The majority of it is type definitions for the various Open Graph objects and theory test cases to check that each other the properties is output properly. The logic is mostly contained in the `OpenGraphTagHelper`, `OpenGraphProperty` and `PropertyInfoExtensions` classes.

File: src/Winton.AspNetCore.Seo/HeaderMetadata/OpenGraph/Music/Album.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        ///     Initializes a new instance of the <see cref="Album"/> class.
+        /// </summary>
+        /// <param name="url">The url of another Open Graph object with og:type music.album.</param>
+        public Album(string url)
+        {
+            Url = url;
+        }
+
+        /// <summary>
+        ///     Gets or sets he disc (within the album) that the song is on, defaulting to to '1'.
```
Comment (richthornton) [inline src/Winton.AspNetCore.Seo/HeaderMetadata/OpenGraph/Music/Album.cs]: Typo `he` should be `the`

File: test/Winton.AspNetCore.Seo.Tests/HeaderMetadata/OpenGraph/OpenGraphPropertyTests.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+                OpenGraphProperty openGraphProperty = OpenGraphProperty.Create(propertyInfo, openGraphTagHelper);
+
+                openGraphProperty.FullName.Should().Be(expected);
+            }
+
+            [Theory]
+            [InlineData(nameof(TestOpenGraphTagHelper.BogStandardProperty), false)]
+            [InlineData(nameof(TestOpenGraphTagHelper.PrimaryProperty), true)]
+            [InlineData(nameof(TestOpenGraphTagHelper.PropertyWithOverridenName), false)]
```
Comment (richthornton) [inline test/Winton.AspNetCore.Seo.Tests/HeaderMetadata/OpenGraph/OpenGraphPropertyTests.cs]: Sp `PropertyWithOverridenName` you're missing a d

**Review by richthornton — CHANGES_REQUESTED** [review]
Small spelling changes

File: src/Winton.AspNetCore.Seo/HeaderMetadata/OpenGraph/Music/Album.cs (hunk 1 above)
Comment (Choc13) [reply src/Winton.AspNetCore.Seo/HeaderMetadata/OpenGraph/Music/Album.cs]: Fixed

File: test/Winton.AspNetCore.Seo.Tests/HeaderMetadata/OpenGraph/OpenGraphPropertyTests.cs (hunk 2 above)
Comment (Choc13) [reply test/Winton.AspNetCore.Seo.Tests/HeaderMetadata/OpenGraph/OpenGraphPropertyTests.cs]: Fixed

**Review by richthornton — APPROVED** [review]

**Pull request merged**